pkgs/development/lua-modules/lib.nix

```diff
@@ -22,13 +22,8 @@ rec {
   inherit overrideLuarocks;
   inherit hasLuaModule requiredLuaModules;
 
-  luaPathList = [
-    "share/lua/${lua.luaversion}/?.lua"
-    "share/lua/${lua.luaversion}/?/init.lua"
-  ];
-  luaCPathList = [
-    "lib/lua/${lua.luaversion}/?.so"
-  ];
+  luaPathList = lua.LUA_PATH_PATTERN;
+  luaCPathList = lua.LUA_CPATH_PATTERN;
```

Comment:
do we even know what there are used for?  at the very least there should be a comment explaining their purpose.

Reply:
agreed. This helps discriminate what package to bring into LUA_PATH (otherwise LUA_PATH grows too big) but it was buggy and that commit didn't improve the situation. I've removed the commit, it was buggy and added to confusion. Let's do a better thing in a future PR, for instance just add derivation to LUA_PATH if derivation contains lua files.

Reply:
> This helps discriminate what package to bring into LUA_PATH

no, `lua*.LuaPathSearchPaths` does that, or more precisely, the arguments to luaPackages.lua-setup-hook.

> I've removed the commit, it was buggy and added to confusion.

what commit? this was already part of the functionality of nixpkgs.

> for instance just add derivation to LUA_PATH if derivation contains lua files.

this seems like a horrible idea.  you'd eventually end up with serious namespace pollution issues where you're importing the guts of some lua program into your local namespace.

there's a reason the envHook prefix system exists.